### PR TITLE
add a clean up list of languages

### PR DESCRIPTION
**Pull request by Fahad-Alsaidi**
I tried to clean up many entries as possible, please review before merge ;-)

### CODE REVIEW

File: scribus/langmgr.cpp [hunk 1]
```diff
@@ -89,90 +89,545 @@ void LanguageManager::generateLangList()
...
-	m_langTable.append(LangDef("uk",       "uk_UA",    "Ukranian",                QObject::tr("Ukranian")));
-	m_langTable.append(LangDef("ur",       "",         "Urdu",                    QObject::tr("Urdu")));
-	m_langTable.append(LangDef("vi",       "",         "Vietnamese",              QObject::tr("Vietnamese")));
-	m_langTable.append(LangDef("cy",       "cy_GB",    "Welsh",                   QObject::tr("Welsh")));
+	m_langTable.append(LangDef("aiw",      "",         "Aari",                                QObject::tr("Aari")));
+	m_langTable.append(LangDef("abq",      "",         "Abaza",                               QObject::tr("Abaza")));
+	m_langTable.append(LangDef("ab",       "",         "Abkhazian",                           QObject::tr("Abkhazian")));
+	m_langTable.append(LangDef("ach",      "",         "Acoli",                               QObject::tr("Acoli")));
+	m_langTable.append(LangDef("ady",      "",         "Adyghe",                              QObject::tr("Adyghe")));
+	m_langTable.append(LangDef("prs",      "",         "Afghan Persian/Dari",                 QObject::tr("Afghan Persian/Dari")));
```
Comment (khaledhosny) [inline scribus/langmgr.cpp]: “Dari” should be enough.

File: scribus/langmgr.cpp [hunk 2]
```diff
@@ -89,90 +89,545 @@ void LanguageManager::generateLangList()
...
+	m_langTable.append(LangDef("aio",      "",         "Aiton",                               QObject::tr("Aiton")));
+	m_langTable.append(LangDef("ak",       "",         "Akan",                                QObject::tr("Akan")));
+	m_langTable.append(LangDef("sq",       "",         "Albanian",                            QObject::tr("Albanian")));
+	m_langTable.append(LangDef("atv",      "alt",      "Altai",                               QObject::tr("Altai")));
+	m_langTable.append(LangDef("am",       "",         "Amharic",                             QObject::tr("Amharic")));
+	m_langTable.append(LangDef("ar",       "",         "Arabic",                              QObject::tr("Arabic")));
+	m_langTable.append(LangDef("an",       "an_ES",    "Aragonese",                           QObject::tr("Aragonese")));
+	m_langTable.append(LangDef("hy",       "",         "Armenian",                            QObject::tr("Armenian")));
+	m_langTable.append(LangDef("rup",      "",         "Aromanian/Arumanian/Macedo-Romanian", QObject::tr("Aromanian/Arumanian/Macedo-Romanian")));
```
Comment (khaledhosny) [inline scribus/langmgr.cpp]: “Aromanian” should be enough.

File: scribus/langmgr.cpp [hunk 3]
```diff
@@ -89,90 +89,545 @@ void LanguageManager::generateLangList()
...
+	m_langTable.append(LangDef("frp",      "",         "Arpitan/Francoprovençal",             QObject::tr("Arpitan/Francoprovençal")));
+	m_langTable.append(LangDef("as",       "",         "Assamese",                            QObject::tr("Assamese")));
```
Comment (khaledhosny) [inline scribus/langmgr.cpp]: Just choose one of the two names here, either “Arpitan” or “Francoprovençal”. This should happen for all languages with multiple names on the list, just pick the most common one (e.g. the Wikipedia article name for that language).

**Comment by Fahad-Alsaidi** [comment]
updated.

File: scribus/langmgr.cpp [hunk 4]
```diff
@@ -89,90 +89,541 @@ void LanguageManager::generateLangList()
...
+	m_langTable.append(LangDef("eve",      "",         "Even",                                QObject::tr("Even")));
+	m_langTable.append(LangDef("evn",      "",         "Evenki",                              QObject::tr("Evenki")));
+	m_langTable.append(LangDef("ee",       "",         "Ewe",                                 QObject::tr("Ewe")));
+	m_langTable.append(LangDef("fo",       "",         "Faroese",                             QObject::tr("Faroese")));
+	m_langTable.append(LangDef("fj",       "",         "Fijian",                              QObject::tr("Fijian")));
+	m_langTable.append(LangDef("fil",      "",         "Filipino",                            QObject::tr("Filipino")));
+	m_langTable.append(LangDef("fi",       "",         "Finnish",                             QObject::tr("Finnish")));
+	m_langTable.append(LangDef("fon",      "",         "Fon",                                 QObject::tr("Fon")));
+	m_langTable.append(LangDef("myq",      "",         "Forest Maninka",                      QObject::tr("Forest Maninka")));
+	m_langTable.append(LangDef("frp",      "",         "Franco-Provençal",                     QObject::tr("Franco-Provençal")));
```
Comment (khaledhosny) [inline scribus/langmgr.cpp]: I think `QObject::tr()` does not work well with non-ASCII characters, you need to escape them (see Bokmål below).

File: scribus/langmgr.cpp [hunk 5]
```diff
@@ -89,90 +89,541 @@ void LanguageManager::generateLangList()
...
+	m_langTable.append(LangDef("rw",       "",         "Kinyarwanda",                         QObject::tr("Kinyarwanda")));
+	m_langTable.append(LangDef("mwk",      "",         "Kita Maninkakan",                     QObject::tr("Kita Maninkakan")));
+	m_langTable.append(LangDef("kex",      "",         "Kituba (Congo)",                      QObject::tr("Kituba (Congo)")));
+	m_langTable.append(LangDef("kex",      "",         "Kokni",                               QObject::tr("Kokni")));
+	m_langTable.append(LangDef("ksh",      "",         "Kölsch",                              QObject::tr("Kölsch")));
+	m_langTable.append(LangDef("kv",       "",         "Komi",                                QObject::tr("Komi")));
+	m_langTable.append(LangDef("koi",      "",         "Komi-Permyak",                        QObject::tr("Komi-Permyak")));
+	m_langTable.append(LangDef("kpv",      "",         "Komi-Zyrian",                         QObject::tr("Komi-Zyrian")));
+	m_langTable.append(LangDef("kmw",      "",         "Komo (Democratic Republic of Congo)", QObject::tr("Komo (Democratic Republic of Congo)")));
```
Comment (khaledhosny) [inline scribus/langmgr.cpp]: I think “Congo” instead of the long “Democratic Republic of Congo” should be enough, as done above already.

**Comment by Fahad-Alsaidi** [comment]
updated

**Comment by aoloe** [comment]
wow! that's going on holidays all around the world :-)

**Comment by khaledhosny** [comment]
OK, looks fine now. One more change and I’ll merge it: can you change the spacing so that the diff shows only the newly added entries, right now the diff is not very helpful.

**Comment by Fahad-Alsaidi** [comment]
updated. However, to do that I shorten some languages names. As you see from the diff you can notice that I delete some old entries because some of them have incorrect iso code and the other are just duplicated.

**Comment by khaledhosny** [comment]
You can keep the long language names unaligned with the rest, no need to strictly align all the entries.

**Comment by Fahad-Alsaidi** [comment]
ok. updated.

**Pull request merged**